### PR TITLE
refactor: improve utils code

### PR DESCRIPTION
**Pull request by BlackHole1**
Improve readability and maintainability of code

### CODE REVIEW

**Review by VerteDinde — APPROVED** [review]

File: src/utils/electron-name.ts [hunk 1]
```diff
@@ -6,9 +6,11 @@
...
-  } else if (process.platform === 'darwin') {
+  }
+
+  if (process.platform === 'darwin') {
     return 'Electron.app';
-  } else {
-    return 'electron';
   }
+
+  return 'electron';
```
Comment (malept) [inline src/utils/electron-name.ts:15]: suggestion: this could just be a switch statement.

File: src/utils/exec.ts [hunk 2]
```diff
@@ -23,28 +46,11 @@ export async function exec(dir: string, cliArgs: string): Promise<string> {
...
         if (error) {
-          reject(error);
+          return reject(error);
         }
 
-        resolve(typeof result === 'string' ? result : `${result}`);
+        resolve(String(result));
       },
     );
   });
```
Comment (malept) [inline src/utils/exec.ts:55]: suggestion: if we're generally refactoring for clarity in this PR, it seems like you could just `promisify` the exec call.

File: src/utils/gist.ts [hunk 3]
```diff
@@ -13,17 +13,9 @@
...
-  if (input.startsWith('https://gist.github.com')) {
-    if (input.endsWith('/')) {
-      input = input.slice(0, -1);
-    }
-    id = input.split('/').pop()!;
-  }
-
-  return id.match(/[0-9A-Fa-f]{32}/) ? id : null;
+  return id?.[0] || null;
 }
```
Comment (malept) [inline src/utils/gist.ts:19]: question: do we have unit tests for this function?

Comment (BlackHole1) [reply src/utils/gist.ts:19]: Yes. Here: https://github.com/electron/fiddle/blob/3814ec9be6e432eca26eaade778efd15d0968c48/tests/utils/gist-spec.ts#L41-L85

**Comment by BlackHole1** [comment]
@malept Done. Please re-review

File: src/utils/gist.ts (hunk 3 above)
Comment (malept) [reply src/utils/gist.ts:19]: It'd be nice if the null case(s) were covered in the testsuite.

Comment (BlackHole1) [reply src/utils/gist.ts:19]: Good idea.

Comment (BlackHole1) [reply src/utils/gist.ts:19]: Done

**Pull request merged**